devtools/maven/src/main/java/io/quarkus/maven/RunMojo.java

```diff
@@ -0,0 +1,112 @@
+package io.quarkus.maven;
+
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import io.quarkus.bootstrap.app.AugmentAction;
+import io.quarkus.bootstrap.app.CuratedApplication;
+import io.quarkus.deployment.run.RunCommandHandler;
+import io.quarkus.deployment.run.RunCommandLaunchResultBuildItem;
+
+@Mojo(name = "run")
+public class RunMojo extends QuarkusBootstrapMojo {
+    /**
+     * The list of system properties defined for the plugin.
+     */
+    @Parameter
+    Map<String, String> systemProperties = Collections.emptyMap();
+
+    @Override
+    protected boolean beforeExecute() throws MojoExecutionException, MojoFailureException {
+        return true;
+    }
+
+    @Override
+    protected void doExecute() throws MojoExecutionException, MojoFailureException {
```

Comment:
the execution have  what looks like a copy of all system properties on it. is that intended?

`-Djava.specification.version=17 -Dsun.jnu.encoding=UTF-8 -Djava.class.path=/Users/manderse/temp/myj/.mvn/wrapper/maven-wrapper.jar -Djava.vm.vendor=GraalVM Community -Dsun.arch.data.model=64 -Djava.vendor.url=https://www.graalvm.org/ -Duser.timezone=Europe/Copenhagen -Dmaven.conf=/Users/manderse/.m2/wrapper/dists/apache-maven-3.8.7-bin/678cc9d4/apache-maven-3.8.7/conf -Djava.vm.specification.version=17 -Dos.name=Mac OS X -Dsun.java.launcher=SUN_STANDARD -Duser.country=DK -Dsun.boot.library.path=/Users/manderse/.sdkman/candidates/java/22.3.r17-grl/lib -Dsun.java.command=org.apache.maven.wrapper.MavenWrapperMain -f /Users/manderse/temp/myj/pom.xml verify -DskipTests quarkus:run -Dhttp.nonProxyHosts=local|*.local|169.254/16|*.169.254/16 -Djdk.debug=release -Dmaven.home=/Users/manderse/.m2/wrapper/dists/apache-maven-3.8.7-bin/678cc9d4/apache-maven-3.8.7 -Dsun.cpu.endian=little -Duser.home=/Users/manderse -Duser.language=en -Dsun.stderr.encoding=UTF-8 -Djava.specification.vendor=Oracle Corporation -Djava.version.date=2022-10-18 -Djava.home=/Users/manderse/.sdkman/candidates/java/22.3.r17-grl -Dfile.separator=/ -Djava.vm.compressedOopsMode=Zero based -Djdk.internal.vm.ci.enabled=true -Dline.separator=
 -Dsun.stdout.encoding=UTF-8 -Djava.vm.specification.vendor=Oracle Corporation -Djava.specification.name=Java Platform API Specification -Dapple.awt.application.name=MavenWrapperMain -Dsun.management.compiler=HotSpot 64-Bit Tiered Compilers -Dftp.nonProxyHosts=local|*.local|169.254/16|*.169.254/16 -Djava.runtime.version=17.0.5+8-jvmci-22.3-b08 -Duser.name=manderse -Dpath.separator=: -Dos.version=13.2.1 -Djava.runtime.name=OpenJDK Runtime Environment -Dfile.encoding=UTF-8 -Dguice.disable.misplaced.annotation.check=true -Djava.vm.name=OpenJDK 64-Bit Server VM -DskipTests=true -Djava.vendor.version=GraalVM CE 22.3.0 -Djava.vendor.url.bug=https://github.com/oracle/graal/issues -Djava.io.tmpdir=/var/folders/mm/z7zzmyl15bd8byr8vsdxf1740000gn/T/ -Djava.version=17.0.5 -Duser.dir=/Users/manderse/temp/myj -Dos.arch=aarch64 -Dmaven.multiModuleProjectDirectory=/Users/manderse/temp/myj -Djava.vm.specification.name=Java Virtual Machine Specification -Dnative.encoding=UTF-8 -Djava.library.path=/Users/manderse/Library/Java/Extensions:/Library/Java/Extensions:/Network/Library/Java/Extensions:/System/Library/Java/Extensions:/usr/lib/java:. -Djava.vm.info=mixed mode, sharing -Djava.vendor=GraalVM Community -Dvertx.disableTCCL=true -Djava.vm.version=17.0.5+8-jvmci-22.3-b08 -Dclassworlds.conf=/Users/manderse/.m2/wrapper/dists/apache-maven-3.8.7-bin/678cc9d4/apache-maven-3.8.7/bin/m2.conf -Dsun.io.unicode.encoding=UnicodeBig -DsocksNonProxyHosts=local|*.local|169.254/16|*.169.254/16 -Djava.class.version=61.0 -jar /Users/manderse/temp/myj/target/quarkus-app/quarkus-run.jar`
 
 ?

Reply:
Do you think its a good idea to have a separate switch for passing sys props to spawned process?